### PR TITLE
ci: Move slow lint to lint-slow.sh

### PR DESCRIPTION
**Pull request by def-**
Seen in https://buildkite.com/materialize/test/builds/92033#019273fc-0709-4cb2-b4ff-f11da7bc9ef2

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered. ([trigger-ci for additional test/nightly runs](https://trigger-ci.dev.materialize.com/))
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] If this PR includes major [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note), I have pinged the relevant PM to schedule a changelog post.

### CODE REVIEW

File: ci/test/pipeline.template.yml [hunk 1]
```diff
@@ -137,7 +137,7 @@ steps:
         inputs:
           - "*"
         depends_on: []
-        timeout_in_minutes: 10
+        timeout_in_minutes: 20
```
Comment (nrainer-materialize) [inline ci/test/pipeline.template.yml:140]: The step is called lint-fast 😀

**Review by nrainer-materialize — APPROVED** [review]

File: ci/test/pipeline.template.yml (hunk 1 above)
Comment (def-) [reply ci/test/pipeline.template.yml:140]: I want to see what is taking that long, I have a guess already

**Comment by def-** [comment]
As I expected: `[531.14s] check-crate-license-listing.sh`
Same for me locally, I think we should move that check somewhere else, will do so.

**Review by nrainer-materialize — APPROVED** [review]

**Comment by nrainer-materialize** [comment]
Is this PR obsolete with https://github.com/MaterializeInc/materialize/pull/29937?

**Pull request merged**